test/pico_divider_test/pico_divider_nesting_test.c

```diff
@@ -158,6 +158,23 @@ void test_nesting() {
             if (0xffffffffu != a * 11 + b) {
                 FAILED();
             }
+            static uint64_t z64;
+            z64 -= 47;
+            uint64_t a64 = z64 / -13635;
+            uint64_t b64 = z64 % -13635;
+            if (z64 != a64 * -13635 + b64) {
+                FAILED();
+            }
+            // specifically check 64/32 divide
+            static uint64_t c64 = 0x13ffffffffull;
+            static uint32_t cd = 1;
```

Comment:
I dunno if it'd be "nicer" (more readable?) to declare these outside the `for` loop (like you did with `uint z = 0;`), rather than declaring them as `static` inside the `for` loop? :shrug: 

The first time I saw
```
static uint32_t cd = 1;
a64 = c64 / cd;
```
I thought "Huh, what's the point of testing divide-by-one?" :wink:  (it was only later that I spotted the `cd++;` further down)

(but then I'm not a regular C programmer, so perhaps this is just a common idiom that I'm not familiar with?)

Reply:
yeah i thought about that when i wrote it, but didn't feel motivated enough then, and so probably less so now... these tests are a bit opaque anyway